server/src/main/java/model/shooting/board/ShootingBoard.java

```diff
@@ -0,0 +1,138 @@
+package model.shooting.board;
+
+import model.shooting.field.HitResult;
+import model.shooting.ship.ShootingShip;
+import model.shooting.field.Field;
+import model.shooting.field.FieldStateName;
+import model.Coordinates;
+import model.placement.fleet.Fleet;
+import model.placement.ship.PlacedShip;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.HashSet;
+
+public class ShootingBoard implements ObservableBoard {
+    private static final int COLUMNS_NUMBER = 10;
+    private static final int ROWS_NUMBER = 10;
+    private Map<Coordinates, Field> coordinatesFieldMap;
+    private List<BoardObserver> observers;
+    private final int rows;
+    private final int cols;
+
+    private ShootingBoard(Map<Coordinates, Field> coordinatesFieldMap, int rows, int cols) {
+        this.coordinatesFieldMap = coordinatesFieldMap;
+        this.observers = new ArrayList<>();
+        this.rows = rows;
+        this.cols = cols;
+    }
+
+    public static ShootingBoard createEmpty(int rows, int cols) {
+        return new ShootingBoard(new HashMap<>(), rows, cols);
+    }
+
+    public static ShootingBoard fromFleet(Fleet<PlacedShip> placedShipFleet) {
+        HashMap<Coordinates, Field> coordinatesFieldHashMap = new HashMap<>();
+        placedShipFleet.getShipList().
+                forEach(ship -> ship.getMastCoordinates().
+                        forEach(coordinates -> coordinatesFieldHashMap.put(coordinates, Field.withShip())));
+        ShootingBoard shootingBoard = new ShootingBoard(coordinatesFieldHashMap, ROWS_NUMBER, COLUMNS_NUMBER);
+
+        placedShipFleet.getShipList().
+                forEach(ship -> shootingBoard.addObserver(new ShootingShip(ship.getMastCoordinates())));
+        return shootingBoard;
+    }
+
+    public HitResult hit(Coordinates coordinates) {
+        if (isCoordinateOutOfBoard(coordinates)) {
+            return HitResult.OUT_OF_BOARD;
+        }
+        Field field = coordinatesFieldMap.getOrDefault(coordinates, Field.empty());
+        HitResult hitResult = field.hit();
+        coordinatesFieldMap.put(coordinates, field);
+
+        if (!hitResult.equals(HitResult.MISS)) {
```

Comment:
equlas to method -> isNotMiss, or sth like that

Reply:
Removed.